### PR TITLE
Add options for passing in Closure Template compiler JAR and utils JS

### PR DESCRIPTION
**Pull request by bregenspan**
This removes the hard dependency on Medium's packaged version of closure-templates and adds two new options:

* soyJarDir (path to Closure template compiler JAR)
* soyUtilsPath (path to soyutils JS file that should be injected into the VM)

These allow us to target the version we're using, which currently lies somewhere in between Medium's packaged versions.

### CODE REVIEW

**Comment by bregenspan** [comment]
@brycedorn would you mind taking a look at this? Straightforward, just want to make sure there's another set of eyes on this. Hopefully will have PR up on Mantle side later today with some POC of its use.

File: lib/SoyOptions.js [hunk 1]
```diff
@@ -21,6 +21,18 @@ function SoyOptions() {
   this.inputDir = process.cwd()
 
   /**
+   * Directory containing the Closure Templates SoyToJsSrcCompiler.jar
+   * @type {string}
+   */
+  this.soyJarDir = process.cwd()
```
Comment (brycedorn) [inline lib/SoyOptions.js:27]: should this be `soyJarPath`?

Comment (bregenspan) [reply lib/SoyOptions.js:27]: If passed in as an argument, it gets checked dynamically at:

https://github.com/gawkermedia/soynode/pull/1/files#diff-77bf383a42fff8dd7d74b595e9c73d95L180

And then on consumer side, the path is defined like this (though hopefully without need for the hardcoded node_modules path):
https://github.com/gawkermedia/kinja-mantle/commit/f47e6e6d0c840d26c8dcd5dd1715a1a07f84695a#diff-f0807c322f54cb38e739cc904149f38bR34

Comment (bregenspan) [reply lib/SoyOptions.js:27]: (not seeing a way to require arguments on the SoyOptions side, but the command will fail out quickly if an invalid path is specified)

Comment (bregenspan) [reply lib/SoyOptions.js:27]: Sorry, I see what you meant! Just fixed related typo...

**Pull request merged**